controllers/postgres_controller.go

```diff
@@ -716,11 +716,47 @@ func (r *PostgresReconciler) updatePatroniConfig(ctx context.Context, instance *
 		return err
 	}
 	if len(pods.Items) == 0 {
-		r.Log.Info("no leader pod ready, requeuing")
-		// TODO return proper error
-		return errors.New("no leader pods found")
+		r.Log.Info("no leader pod found, selecting all spilo pods as a last resort (might be ok if it is still creating)")
+
+		opts := []client.ListOption{
+			client.InNamespace(instance.ToPeripheralResourceNamespace()),
+			client.MatchingLabels{"application": "spilo"},
+		}
+		if err := r.SvcClient.List(ctx, pods, opts...); err != nil {
+			r.Log.Info("could not query pods, requeuing")
+			return err
+		}
+
+		if len(pods.Items) == 0 {
+			r.Log.Info("no spilo pods found at all, requeueing")
+			return errors.New("no spilo pods found at all")
+		}
+
+		// iterate all spilo pods
+		var lastErr error = nil
+		for _, pod := range pods.Items {
+			pod := pod // pin!
+			podIP := pod.Status.PodIP
+			if err = r.httpPatchPatroni(ctx, instance, podIP); err != nil {
```

Comment:
Sure there should not be any break condition in the loop? This could trigger many successful patches?

Reply:
this is a las resort measure. usually, there should be a leader pod (either `spilo-role=master` or `spilo-role=standby_leader`), but when shtf, I want to make sure everyone gets the memo :-)